packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -365,57 +354,20 @@ class _DefaultPub implements Pub {
     String? flutterRootOverride,
   }) async {
     int exitCode;
-    int attempts = 0;
-    int duration = 1;
-
-    List<_OutputLine>? output;
-    StreamSubscription<String> recordLines(Stream<List<int>> stream, _OutputStream streamName) {
-      return stream
-        .transform<String>(utf8.decoder)
-        .transform<String>(const LineSplitter())
-        .listen((String line) => output!.add(_OutputLine(line, streamName)));
-    }
 
-    final Status? status = printProgress
-      ? _logger.startProgress('Running "flutter pub $command" in ${_fileSystem.path.basename(directory)}...',)
-      : null;
     final List<String> pubCommand = _pubCommand(arguments);
     final Map<String, String> pubEnvironment = await _createPubEnvironment(context, flutterRootOverride);
     try {
-      do {
-        output = <_OutputLine>[];
-        attempts += 1;
-        final io.Process process = await _processUtils.start(
-          pubCommand,
-          workingDirectory: _fileSystem.path.current,
-          environment: pubEnvironment,
-        );
-        final StreamSubscription<String> stdoutSubscription =
-          recordLines(process.stdout, _OutputStream.stdout);
-        final StreamSubscription<String> stderrSubscription =
-          recordLines(process.stderr, _OutputStream.stderr);
-
-        exitCode = await process.exitCode;
-        unawaited(stdoutSubscription.cancel());
-        unawaited(stderrSubscription.cancel());
-
-        if (retry && exitCode == _kPubExitCodeUnavailable) {
-          _logger.printStatus(
-            '$failureMessage (server unavailable) -- attempting retry $attempts in $duration '
-            'second${ duration == 1 ? "" : "s"}...',
-          );
-          await Future<void>.delayed(Duration(seconds: duration));
-          if (duration < 64) {
-            duration *= 2;
-          }
-          // This will cause a retry.
-          output = null;
-        }
-      } while (output == null);
-      status?.stop();
+      final io.Process process = await _processUtils.start(
+        pubCommand,
+        workingDirectory: _fileSystem.path.current,
+        environment: pubEnvironment,
+        mode: ProcessStartMode.inheritStdio,
```

Comment:
@Jasguerrero did you say there was a bug with `ProcessStartMode.inerheritStdio` on Windows?

Reply:
I tried this on Windows and at least I saw the stdout. Not sure if there were other gotchas you knew about

Reply:
> @Jasguerrero did you say there was a bug with `ProcessStartMode.inerheritStdio` on Windows?

I didn't try specifically on windows but it was not working for me on Mac

Reply:
Oh, huh. Could it be that you were effected by this bug?

https://github.com/flutter/flutter/pull/115801/commits/fb55f91d7d22c098d1c4db2d8476564cb1e078b6

`ErrorHandlingProcessManager` was silently eating the `mode` parameter instead of forwarding it to the backing `ProcessManager`. I found it when trying to get inheritStdio to work. After that, I was able to get it working on my Mac.

Reply:
nice find!